### ISSUE TITLE
dotnet processes stay alive after running any commands via cmd - on Windows 11 - .NET8

### ISSUE THREAD

**Issue by MichaCo** [dotnet/sdk#42868]
I have the latest .NET8 SDK installed
![image](https://github.com/user-attachments/assets/ac606f9a-7e86-4ec1-8d22-523134df1cd1)

Windows 11
![image](https://github.com/user-attachments/assets/f2f51874-0cb0-4ffd-921a-b14566d57733)

When I `dotnet restore` or `dotnet build` any solution via command line (not from within an IDE), I always see the following processes hanging in task manager:

![image](https://github.com/user-attachments/assets/a75f6ac6-2544-4068-821c-f39fca33a639)

### Bug

The processes stay there forever and do not terminate even after closing the console I started them from.

### Expected behavior

The processes end and free up memory after the `dotnet` command ran.

### Woraround

I always have a script handy to kill the garbage and free memory:

```cmd
taskkill /F /IM dotnet.exe
```

---

If there are logs in some place you want me to attach, please point me to it, I don't know if there is anything which would help

Thanks,
M

**Comment by baronfel** [comment]
This is currently behaving as designed - MSBuild worker nodes, razor compiler server, and Roslyn compiler server by design keep long-lived processes (~15m lifetimes) around to reduce the time it takes to perform the _next_ build.

**Comment by MichaCo** [comment]
Thanks for the reply @baronfel, I did not know that. If that's well documented then my bad.

That being said
If I do builds in Visual Studio, and close VS, it also terminates all processes VS started. 
Wouldn't it be better if other hosts would also kill the processes they start, IF you close the host? Exactly like VSCode, and other tools. If I close VSCode, I'd also expect that all processes used are terminated, no matter what that is, node.js,... dotnet...

**Comment by baronfel** [comment]
Yes - you can run `dotnet build-server shutdown` to have the CLI shut down the daemons that it has spawned. If there are scenarios where DevKit doesn't tear down its instances, I would encourage you to raise an issue at https://github.com/microsoft/vscode-dotnettools/issues!

**Comment by baronfel** [comment]
@MichaCo I think I'd like to close this issue as a duplicate of https://github.com/dotnet/sdk/issues/20183, which is the core problem here. We need MSBuild to do process tracking before we can implement a 'one-shot' mode for the CLI.

**Comment by MichaCo** [comment]
sounds good
